amp-beta-tester.php

```diff
@@ -0,0 +1,292 @@
+	}
+
+	return $updates;
+}
+
+/**
+ * Update the AMP plugin information data to reflect that of the GitHub release.
+ *
+ * @param false|object|array $value  The result object or array. Default false.
+ * @param string             $action The type of information being requested from the Plugin Installation API.
+ * @param object             $args   Plugin API arguments.
+ * @return false|object|array Updated $value, or passed-through $value on failure.
+ */
+function update_amp_plugin_information( $value, $action, $args ) {
+	if ( 'plugin_information' !== $action || 'amp' !== $args->slug ) {
+		return $value;
+	}
+
+	$amp_version = get_amp_version();
+
+	if ( ! is_pre_release( $amp_version ) ) {
+		return $value;
+	}
+
+	$amp_manifest = get_amp_update_manifest( $amp_version );
+
+	if ( ! $amp_manifest ) {
+		return $value;
+	}
+
+	$value->version = $amp_version;
+
+	$value->download_link            = $amp_manifest->package;
+	$value->versions[ $amp_version ] = $amp_manifest->package;
+
+	return $value;
+}
+
+/**
+ * Fetch AMP releases from GitHub.
+ *
+ * @return array|false
+ */
+function get_amp_github_releases() {
+	$raw_response = wp_remote_get( 'https://api.github.com/repos/ampproject/amp-wp/releases' );
+	if ( is_wp_error( $raw_response ) ) {
+		return false;
+	}
+	return json_decode( $raw_response['body'] );
+}
+
+/**
+ * Retrieves the download url for amp.zip, if it exists.
+ *
+ * @param object $release GitHub release JSON object.
+ * @return string|false Download URL if it exists, false if not.
+ */
+function get_download_url_from_amp_release($release ) {
+	foreach ( $release->assets as $asset ) {
+		if ( 'amp.zip' === $asset->name ) {
+			return $asset->browser_download_url;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Retrieves the current AMP update manifest, and updates it to include the analogous information
+ * from its GitHub release.
+ *
+ * @param object $release GitHub release JSON object.
+ * @return array|false Updated manifest, or false if it fails to retrieve the current update manifest.
+ */
+function generate_amp_update_manifest($release ) {
+	$current_manifest = get_current_amp_update_manifest();
+
+	if ( ! $current_manifest ) {
+		return false;
+	}
+
+	$manifest = [];
+	$zip_url  = get_download_url_from_amp_release( $release );
+
+	if ( $zip_url ) {
+		$manifest['package'] = $zip_url;
+	}
+
+	$manifest['new_version'] = $release->tag_name;
+	$manifest['url']         = $release->html_url;
+
+	return array_merge( (array) $current_manifest, $manifest );
+}
+
+/**
+ * Get the AMP plugin update manifest for the specified version from GitHub.
+ *
+ * @param string $version Version to get manifest for. Defaults to getting the latest pre-release.
+ * @return object|false Latest release, or false on failure.
+ */
+function get_amp_update_manifest($version = 'pre-release' ) {
+	$amp_manifest = null;
+	$releases     = get_site_transient( AMP__BETA__TESTER__RELEASES__TRANSIENT );
+
+	if ( empty( $releases ) ) {
```

Comment:
See comment above regarding storing the transient.
```suggestion
	if ( false === $releases ) {
```